TypeSystem/variant.h

```diff
@@ -304,21 +307,25 @@ struct VariantImpl<TypeListImpl<TYPES...>> : CurrentVariant {
 };
 
 // `Variant<...>` can accept either a list of types, or a `TypeList<...>`.
-template <typename T, typename... TS>
+template <typename NAME, typename T, typename... TS>
```

Comment:
Can't we just use `typename... TS` instead of `typename T, typename... TS` here?